scalabel/eval/detect.py

```diff
@@ -0,0 +1,282 @@
+            (n_thr, n_area, n_mdet)
+        )
+
+    # Print evaluation results
+    stats = np.zeros((n_tit, 1))
+    print("\nOverall performance")
+    coco_eval.eval = eval_param
+    coco_eval.summarize()
+
+    for i in range(n_tit):
+        column = stats_all[:, i]
+        if len(column > -1) == 0:
+            stats[i] = -1
+        else:
+            stats[i] = np.mean(column[column > -1], axis=0)
+
+    score_titles = [
+        "AP",
+        "AP_50",
+        "AP_75",
+        "AP_small",
+        "AP_medium",
+        "AP_large",
+        "AR_max_1",
+        "AR_max_10",
+        "AR_max_100",
+        "AR_small",
+        "AR_medium",
+        "AR_large",
+    ]
+    scores: Dict[str, float] = {}
+
+    for title, stat in zip(score_titles, stats):
+        scores[title] = stat.item()
+
+    if out_dir != "none":
+        write_eval(out_dir, scores, eval_param)
+    return scores
+
+
+def write_eval(
+    out_dir: str, scores: Dict[str, float], eval_param: DictStrAny
+) -> None:
+    """Write the evaluation results to file, print in tabulate format."""
+    output_filename = os.path.join(out_dir, "scores.json")
+    with open(output_filename, "w") as fp:
+        json.dump(scores, fp)
+
+    # print the overall performance in the tabulate format
+    print(create_small_table(scores))
+
+    eval_param["precision"] = eval_param["precision"].flatten().tolist()
+    eval_param["recall"] = eval_param["recall"].flatten().tolist()
+
+    with open(os.path.join(out_dir, "eval.json"), "w") as fp:
+        json.dump(eval_param, fp)
+
+
+def create_small_table(small_dict: Dict[str, float]) -> str:
+    """Create a small table using the keys of small_dict as headers.
+
+    Args:
+        small_dict (dict): a result dictionary of only a few items.
+
+    Returns:
+        str: the table as a string.
+    """
+    keys, values_t = tuple(zip(*small_dict.items()))
+    values = ["{:.1f}".format(val * 100) for val in values_t]
+    stride = 3
+    items: ListAny = []
+    for i in range(0, len(keys), stride):
+        items.append(keys[i : min(i + stride, len(keys))])
+        items.append(values[i : min(i + stride, len(keys))])
+    table = tabulate(
+        items[1:],
+        headers=items[0],
+        tablefmt="fancy_grid",
+        floatfmt=".3f",
+        stralign="center",
+        numalign="center",
+    )
+    return table
+
+
+def parse_arguments() -> argparse.Namespace:
+    """Parse the arguments."""
+    parser = argparse.ArgumentParser(description="Detection evaluation.")
+    parser.add_argument(
+        "--gt", "-g", required=True, help="path to detection ground truth"
+    )
+    parser.add_argument(
+        "--result", "-r", required=True, help="path to detection results"
+    )
+    parser.add_argument(
+        "--cfg-path",
+        "-c",
+        default=DEFAULT_LABEL_CONFIG,
+        help="Config path, contains metadata like available categories.",
```

Comment:
Can we be more precise for the config schema?

Reply:
made more concrete